src/components/registration/LogIn.js

```diff
@@ -12,11 +12,21 @@ class LogIn extends Component {
     this.state = {
       email: '',
       password: '',
+      errors: {
+        email: false,
+        password: false,
+      },
+      ValidationErrorMessages: {
```

Comment:
These are constant/set beforehand, so don't need to be in state. They can just be a constant object outside of state. Also, please use either the caps case naming convention because it's some constant info, or use snake case because it's a regular variable (and same for all of the contained keys).

```suggestion
      // outside of this.state
      this.validation_error_messages: {
```